src/lazynameddimsarrays.jl

```diff
@@ -417,4 +417,12 @@ Base.:*(a::SymbolicNamedDimsArray, b::SymbolicNamedDimsArray) = lazy(a) * lazy(b
 Base.:*(a::SymbolicNamedDimsArray, b::LazyNamedDimsArray) = lazy(a) * b
 Base.:*(a::LazyNamedDimsArray, b::SymbolicNamedDimsArray) = a * lazy(b)
 
+function nested_array_to_lazy_multiply(arr)
+    if arr isa AbstractVector
+        return mapreduce(x -> nested_array_to_lazy_multiply(x), *, arr)
+    else
+        return symnameddims.(arr)
+    end
+end
```

Comment:
```suggestion
function contraction_sequence_to_expr(seq)
    if seq isa AbstractVector
        return prod(contraction_sequence_to_expr, seq)
    else
        return symnameddims(seq)
    end
end
```
Some cleanup and rename, and also removing the broadcasting from the second branch (I think that didn't make sense since at that point it is converting a single label?).

Also, I think this function isn't quite right for the `LazyNamedDimsArrays` module, let's move it to `contractnetwork.jl`.